consensus/core/src/core.rs

```diff
@@ -731,6 +741,18 @@ impl Core {
                 ancestors
                     .into_iter()
                     .filter(|block| block.author() != self.context.own_index)
+                    .filter(|block| {
+                        if gc_enabled {
+                            // When GC is enabled we want to include only ancestors that are:
+                            // 1) blocks of the genesis round
+                            // 2) blocks that are parents of the previous round (clock_round - 1)
+                            // 3) blocks that are parents earlier than the previous round, but higher than the gc_round
+                            return block.round() == GENESIS_ROUND
+                                || block.round() == clock_round - 1
```

Comment:
why do we need `block.round() == clock_round - 1` isn't that covered by `block.round() > gc_round` or are you saying you will include a gc'd ancestor if it is from the parent round? That does not seem like a possible scenario

Reply:
Yeah I think that's me getting a bit paranoid with the checks here. We can certainly just do as you said `block.round() > gc_round`.  In fact the end check can just be:

```
if gc_enabled && gc_round > GENESIS_ROUND {
    return block.round() > gc_round;
}
```